Amazon.IonDotnet/Tree/Impl/IonValue.cs

```diff
@@ -265,13 +272,10 @@ public void ClearAnnotations()
         /// Returns true if the value contains such annotation.
         /// </summary>
         /// <param name="text">Annotation text.</param>
-        /// <exception cref="ArgumentNullException">When text is null.</exception>
         public bool HasAnnotation(string text)
         {
-            if (text == null)
-                throw new ArgumentNullException(nameof(text));
-
-            return _annotations != null && _annotations.Any(a => text.Equals(a.Text));
+            return _annotations != null &&
+                (_annotations.Any(a => (a.Text == null && text == null) || text.Equals(a.Text)));

```

Comment:
Nit:
1) Rename a to be more clear about what it is.
2) Since this is a function on the "a", I think a.text.Equals(Text) would be more suitable.

Reply:
Addressed